elx/catalog.py

```diff
@@ -158,3 +158,45 @@ def select(self, streams: Optional[List[str]]) -> "Catalog":
             )
 
         return catalog
+
+    def set_replication_keys(self, replication_keys: Optional[dict]) -> "Catalog":
+        """
+        Set the replication key for streams and updates the catalog.
+
+        Args:
+            keys (Optional[dict]): Dictionary stream replication_key value-pairs.
+            E.g. {"stream_one": "updated_at", "stream_two": "modified_at"}
+
+        Returns:
+            Catalog: A new catalog with updated replication settings.
+        """
+        # Make a copy of the existing catalog.
+        catalog = self.copy(deep=True)
+
+        # Loop over the streams
+        for stream in catalog.streams:
+            # If the stream is specified in `replication_keys` dictionary
+            if stream.tap_stream_id in replication_keys:
+                # Set the replication method to INCREMENTAL
+                stream.replication_method = "INCREMENTAL"
+
+                # Update the replication key value
+                stream.replication_key = replication_keys[stream.tap_stream_id]
+
+                # Add the replication key to stream metadata
+                stream.upsert_metadata(
+                    breadcrumb=[],
+                    metadata={
+                        "valid-replication-keys": [
+                            replication_keys[stream.tap_stream_id]
+                        ]
+                    },
+                )
```

Comment:
Is dit nodig om toe te voegen?

Reply:
Ehh, ik denk niet dat het per se nodig is functioneel gezien maar staat in Singer spec dus dacht wel zo netjes om dat te volgen.

Reply:
Als het werkt zonder zou ik het denk ik weglaten, maakt het weer wat simpeler. Want in theorie zou die replication key er natuurlijk al in moeten staan. 

Voelt meer als een manier voor de tap om te publishen wat er mogelijk is. Maar als je hem al gebruikt is ook valid natuurlijk.